google/cloud/spanner/samples/samples.cc

```diff
@@ -1160,6 +1125,208 @@ void DeleteData(google::cloud::spanner::Client client) {
+}
+//! [END spanner_insert_datatypes_data]
+
+//! [START spanner_query_with_array_parameter]
+void QueryWithArrayParameter(google::cloud::spanner::Client client) {
+  namespace spanner = ::google::cloud::spanner;
+  std::vector<absl::CivilDay> example_array = {absl::CivilDay(2020, 10, 1),
+                                               absl::CivilDay(2020, 11, 1)};
+  spanner::SqlStatement select(
+      "SELECT VenueId, VenueName, AvailableDate FROM Venues v,"
+      " UNNEST(v.AvailableDates) as AvailableDate "
+      " WHERE AvailableDate in UNNEST(@available_dates)",
+      {{"available_dates", spanner::Value(example_array)}});
+  using RowType = std::tuple<std::int64_t, absl::optional<std::string>,
+                             absl::optional<absl::CivilDay>>;
+  auto rows = client.ExecuteQuery(std::move(select));
+  for (auto const& row : spanner::StreamOf<RowType>(rows)) {
+    if (!row) throw std::runtime_error(row.status().message());
+    std::cout << "VenueId: " << std::get<0>(*row) << "\t";
+    std::cout << "VenueName: " << std::get<1>(*row).value() << "\t";
+    std::cout << "AvailableDate: " << std::get<2>(*row).value() << "\n";
+  }
+  std::cout << "Query completed for [spanner_query_with_array_parameter]\n";
+}
+//! [END spanner_query_with_array_parameter]
+
+//! [START spanner_query_with_bool_parameter]
+void QueryWithBoolParameter(google::cloud::spanner::Client client) {
+  namespace spanner = ::google::cloud::spanner;
+  bool example_bool = true;
+  spanner::SqlStatement select(
+      "SELECT VenueId, VenueName, OutdoorVenue FROM Venues"
+      " WHERE OutdoorVenue = @outdoor_venue",
+      {{"outdoor_venue", spanner::Value(example_bool)}});
+  using RowType = std::tuple<std::int64_t, absl::optional<std::string>,
+                             absl::optional<bool>>;
+  auto rows = client.ExecuteQuery(std::move(select));
+  for (auto const& row : spanner::StreamOf<RowType>(rows)) {
+    if (!row) throw std::runtime_error(row.status().message());
+    std::cout << "VenueId: " << std::get<0>(*row) << "\t";
+    std::cout << "VenueName: " << std::get<1>(*row).value() << "\t";
+    std::cout << "OutdoorVenue: " << std::get<2>(*row).value() << "\n";
+  }
+  std::cout << "Query completed for [spanner_query_with_bool_parameter]\n";
+}
+//! [END spanner_query_with_bool_parameter]
+
+//! [START spanner_query_with_bytes_parameter]
+void QueryWithBytesParameter(google::cloud::spanner::Client client) {
+  namespace spanner = ::google::cloud::spanner;
+  spanner::Bytes example_bytes("Hello World 1");
+  spanner::SqlStatement select(
+      "SELECT VenueId, VenueName FROM Venues"
+      " WHERE VenueInfo = @venue_info",
+      {{"venue_info", spanner::Value(example_bytes)}});
+  using RowType = std::tuple<std::int64_t, absl::optional<std::string>>;
+  auto rows = client.ExecuteQuery(std::move(select));
+  for (auto const& row : spanner::StreamOf<RowType>(rows)) {
+    if (!row) throw std::runtime_error(row.status().message());
```

Comment:
The code coverage build says we do not reach this line, which suggests maybe the example is not working as expected?  I have no idea what, the only thing that jumps at me is maybe we need to insert the bytes column using the `spanner::Bytes` wrapper?

Reply:
Right you are.  Thanks for catching that.

[I might say that it is surprising that the `Commit()` succeeds with ill-formatted `BYTES` values.]